src/parser.ts

```diff
@@ -277,7 +277,10 @@ type _takeVarDefinitionRec<Definitions extends any[], In extends any[]> = In ext
   ? _match<Definitions, In>
   : takeVarDefinition<In> extends _match<infer Definition, infer In>
     ? _takeVarDefinitionRec<[...Definitions, Definition], In>
-    : void;
+    : takeValue<In, true> extends _match<infer _, infer In>
```

Comment:
We should be able to split out a new `takeString` here without any performance consequences probably